scripts/extensions/sams/src/components/assets/assetEditorPanel.tsx

```diff
@@ -69,16 +71,59 @@ export class AssetEditorPanelComponent extends React.PureComponent<IProps, IStat
         this.onChange = this.onChange.bind(this);
         this.onSave = this.onSave.bind(this);
         this.onCancel = this.onCancel.bind(this);
+        this.removeTag = this.removeTag.bind(this);
+        this.addTag = this.addTag.bind(this);
     }
 
     onChange<K extends keyof IAssetItem>(field: K, value: IAssetItem[K]) {
-        this.setState((prevState: IState) => ({
-            updates: {
-                ...prevState.updates,
-                [field]: value,
-            },
-            isDirty: true,
-        }));
+        if (field === 'tags') {
```

Comment:
This would be better moved into the `AssetEditor` itself, otherwise this code (and `addTag`, `removeTag`) would need to be duplicated to `UploadAssetModal` and `EditAttachmentModal` as well

Reply:
`assetEditorPanel`, `uploadAssetModal` and `editAttachmentModal`, all have their own custom `onChange` and states, I think I'll have to probably add `addTag` and `removeTag` separately to them all, customised according to their states.